tests/unit/query-cache-test.js

```diff
@@ -192,6 +192,29 @@ module('unit/query-cache', function(hooks) {
     });
   });
 
+  test('.queryURL can accept headers', function(assert) {
+    assert.equal(this.adapterAjax.callCount, 0, 'initial callCount 0');
+
+    this.adapterAjax.returns(
+      resolve({
+        data: {
+          id: 123,
+          type: 'my-synthetic-type',
+        },
+      })
+    );
+
+    return this.queryCache
+      .queryURL('/uwot', { adapterOptions: { headers: { param: 'value' } } })
+      .then(() => {
+        assert.deepEqual(
+          stubCalls(this.adapterAjax),
+          [[this.adapter + '', ['/uwot', 'GET', { headers: { param: 'value' } }]]],
+          'adapter.ajax called with right headers'
+        );
+      });
+  });
```

Comment:
let's add a testing make sure that these still get merged with any headers on `adapter.headers`